src/game.cpp

```diff
@@ -12296,15 +12239,10 @@ bool game::plmove(int dx, int dy)
 
     } else if( u.has_active_bionic("bio_probability_travel") && u.power_level >= 250 ) {
         //probability travel through walls but not water
-        int tunneldist = 0;
+        tripoint dest( x, y, u.posz() );
         // tile is impassable
-        while ((m.move_cost(x + tunneldist * (x - u.posx()), y + tunneldist * (y - u.posy())) == 0) ||
-               // a monster is there
-               ((mon_at(x + tunneldist * (x - u.posx()), y + tunneldist * (y - u.posy())) != -1 ||
-                 // so keep tunneling
-                 npc_at(x + tunneldist * (x - u.posx()), y + tunneldist * (y - u.posy())) != -1) &&
-                // assuming we've already started
-                tunneldist > 0)) {
+        int tunneldist = 0;
+        do {
             //add 1 to tunnel distance for each impassable tile in the line
             tunneldist += 1;
             if (tunneldist * 250 > u.power_level) { //oops, not enough energy! Tunneling costs 250 bionic power per impassable tile
```

Comment:
If I have exact 500 power, I should be able to tunnel 2 squares, this test is now wrong. (The previous check was for the location that was impassable, now it's for a location that *might* be impassable or it might be the end point where the character appears.)

Moving the condition to the end also changes how `dest` is derived from `tunneldist`. The very first iteration checks for `dest.x == x + 1 * (x - u.posx()) == x + x - u.posx()`, assuming that `x == posx + 1` (attempt to move north or whatever the x-axis is), the check is for `dest.x == (posx + 1) + (posx + 1) - posx == posx + 2`, the place at `posx+1` is never checked at all.